velox/functions/lib/SubscriptUtil.h

```diff
@@ -437,27 +453,65 @@ class SubscriptImpl : public exec::Subscript {
     auto rawSizes = baseMap->rawSizes();
     auto rawOffsets = baseMap->rawOffsets();
 
-    // Search the key in each row.
-    rows.applyToSelected([&](vector_size_t row) {
-      size_t mapIndex = mapIndices[row];
-      size_t size = rawSizes[mapIndex];
-      size_t offset = rawOffsets[mapIndex];
+    // Fast path for the case of a single map. It may be constant or dictionary
+    // encoded. Sort map keys, then use binary search.
+    if (baseMap->size() == 1) {
```

Comment:
@laithsakka How does this criteria compare to your method to check identity between batches?

@mbasmanova Laith is using a method to check if map vector between batches are the same.  Your method looks simpler and less error-prone, while his method covers more cases.

Reply:
@Yuhta I'm aware of Laith's work. This optimization is complimentary as it covers maps with complex-type keys while https://github.com/facebookincubator/velox/pull/7191 applies only to maps with primitive type keys.

Reply:
Also, https://github.com/facebookincubator/velox/pull/7191 applies to cases where there are many maps, not just one.

Reply:
@Yuhta @laithsakka Indeed, it should be straightforward to extend this optimization to primitive types. We can also optimize Nested Loop Join to detect the case of single-row build side and use Constant encoding for build-side columns + sort build-side maps. If we do that, we'll have maps coming to element_at already sorted and that will make this optimization even more effective.

Laith, what do you think?

Reply:
@Yuhta @laithsakka There is another pattern where map is a constant defined in the query. In this case it is not produced by the NLJ. We can add logic to always sort constant map literals so that element_at receives maps sorted in that case too. If we do that, we won't need to make element_at stateful and can keep it stateless and relatively simple.